homeassistant/components/systemmonitor/sensor.py

```diff
@@ -477,13 +523,16 @@ class SystemMonitorSensor(CoordinatorEntity[MonitorCoordinator], SensorEntity):
 
     def __init__(
         self,
-        coordinator: MonitorCoordinator,
+        hass: HomeAssistant,
         sensor_description: SysMonitorSensorEntityDescription,
         entry_id: str,
         argument: str = "",
         legacy_enabled: bool = False,
     ) -> None:
         """Initialize the sensor."""
+        coordinator: MonitorCoordinator = sensor_description.coordinator(
+            hass, sensor_description.key, argument
```

Comment:
We don't want to create one coordinator per sensor. Multiple sensors will share the same coordinator but just take a different view of the coordinator data. Eg there are three sensors that should use the same swap memory coordinator.